aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1844,26 +1844,32 @@ void _fill_matrix_powers(Tensor& buffer, const Tensor& a, int num_matrices) {
 
   // fill a^2
   if (2 <= num_matrices - 1) {
-    at::native::matmul(
-      buffer.select(0, 2), // out for a^2
+    // out for a^2
+    auto view_out = buffer.select(0, 2);
+    at::matmul_out(
```

Comment:
why do we need to go through the dispatcher here instead of calling _matmul_impl?

Reply:
I thought that we should always go through the dispatcher? @albanD @zou3519

Reply:
The "real" calls like `mm` have to go through dispatcher, but calling from one composite implicit function to another composite implicit function can be whatever (for all you know, `matmul` can be a helper function that you are calling from another one).

Reply:
Done!